src/ui/verificationGutterStatusView.ts

```diff
@@ -0,0 +1,376 @@
+      [ lvs.AssertionFailed, grayIcon('error_gray') ],
+      [ lvs.AssertionFailedObsolete, grayIcon('error-obsolete_gray') ],
+      [ lvs.AssertionFailedVerifying, grayIcon('error-verifying_gray') ],
+      [ lvs.AssertionVerifiedInErrorContextObsolete, grayIcon('error-range-verified-obsolete_gray.png') ],
+      [ lvs.AssertionVerifiedInErrorContextVerifying, grayIcon('error-range-verified-obsolete_gray.png') ],
+      [ lvs.AssertionVerifiedInErrorContext, grayIcon('error-range-verified-obsolete_gray.png') ],
+      [ lvs.ErrorContext, grayIcon('error-range_gray') ],
+      [ lvs.ErrorContextStart, grayIcon('error-range-start_gray') ],
+      [ lvs.ErrorContextStartObsolete, grayIcon('error-range-start_gray') ],
+      [ lvs.ErrorContextStartVerifying, grayIcon('error-range-start_gray') ],
+      [ lvs.ErrorContextEnd, grayIcon('error-range-end_gray') ],
+      [ lvs.ErrorContextEndObsolete, grayIcon('error-range-end_gray') ],
+      [ lvs.ErrorContextEndVerifying, grayIcon('error-range-end_gray') ],
+      [ lvs.ErrorContextObsolete, grayIcon('error-range-obsolete_gray') ],
+      [ lvs.ErrorContextVerifying, grayIcon('error-range-verifying_gray') ],
+      [ lvs.VerifiedObsolete, grayIcon('verified_gray') ],
+      [ lvs.VerifiedVerifying, grayIcon('verified_gray') ],
+      [ lvs.Verified, grayIcon('verified_gray') ],
+      [ lvs.Verifying, grayIcon('verified_gray') ],
+      [ lvs.ResolutionError, grayIcon('resolution-error') ]
+    ]);
+  }
+
+  public static createAndRegister(context: ExtensionContext, languageClient: DafnyLanguageClient): VerificationGutterStatusView {
+    const instance = new VerificationGutterStatusView(context);
+    context.subscriptions.push(
+      workspace.onDidCloseTextDocument(document => instance.clearVerificationDiagnostics(document.uri.toString())),
+      window.onDidChangeActiveTextEditor(editor => instance.refreshDisplayedVerificationGutterStatuses(editor)),
+      languageClient.onVerificationStatusGutter(params => instance.updateVerificationStatusGutter(params))
+    );
+    return instance;
+  }
+
+  /// Creation of an decoration type
+  private iconOf(context: ExtensionContext, path: string, grayMode: boolean): TextEditorDecorationType {
+    const icon = context.asAbsolutePath(`images/${path}.png`);
+    return window.createTextEditorDecorationType({
+      isWholeLine: true,
+      rangeBehavior: 1,
+      gutterIconPath: icon,
+      overviewRulerColor:
+        grayMode ? (path === 'resolution-error' ? ScrollColor.Error : ScrollColor.Unknown)
+          : path.startsWith('error-range')
+            ? ScrollColor.ErrorRange
+            : path.startsWith('error')
+              ? ScrollColor.Error
+              : path.startsWith('verified')
+                ? ScrollColor.Verified
+                : ScrollColor.Unknown
+    });
+  }
+
+  /// Helper to create decoration types based on the mode and number of images
+  public makeIconAux(grayMode: boolean, context: ExtensionContext):
```

Comment:
I think this function could either be declared either static or outside of the class.